jupyterlite_sphinx/jupyterlite_sphinx.py

```diff
@@ -608,9 +608,9 @@ def run(self):
         return [content_container_node, notebook_container, script_node]
 
 
-def _process_docstring_examples(app, docname, source):
-    source_path = app.env.doc2path(docname)
-    if source_path.endswith(".py"):
+def _process_docstring_examples(app: Sphinx, docname: str, source: List[str]) -> None:
+    source_path: os.PathLike = Path(app.env.doc2path(docname))
```

Comment:
You shouldn't have to convert the output of `app.env.doc2path` into a `Path`. `app.env.doc2path` should return a `_StrPath`, which inherits from `Path` but adds string methods to it.

Reply:
That makes sense. I noticed that they changed it to convert to `pathlib.Path()` in the release candidate for v8 but rolled it back in the final v8 release, trimming it down to a deprecation warning for further removal in v9. We can handle that later.

I'll convert the `_StrPath` object to a string instead based on how other projects seem to have fixed it:

```suggestion
    source_path: str = str(app.env.doc2path(docname))
```

Reply:
I think it's better just to keep it a `Path` and use a `Path` method. This is my read on what the `Sphinx` developers would prefer. It seems like they're trying to be opinionated and say people should be using `pathlib` to handle paths.

Reply:
I see – do you mean I should use `app.env.doc2path(docname)` as-is, and use `Path.suffix` in the following line?

I imagine downstream extension authors like us would agree that using `pathlib` would be broadly acceptable, though – I find it very elegant!

Reply:
Yes, use the output of `app.env.doc2path` directly and use `Path.suffix`.  In `sphinx 9`, the output of `app.env.doc2path` will be a `Path`

Reply:
Addressed in 13a3631b8e3f52e232a910eb284e6597f94894c3.